ui/app/contexts/metametrics.new.js

```diff
@@ -0,0 +1,220 @@
+            // Some routes include these as params.
+            params: omit(params, ['account', 'address']),
+            network,
+            environment_type: environmentType,
+          },
+          context,
+        })
+      } else if (location.pathname !== '/confirm-transaction') {
+        // We have more specific pages for each type of transaction confirmation
+        // The user lands on /confirm-transaction first, then is redirected based on
+        // the contents of state.
+        captureMessage(`${location.pathname} would have issued a page track event to segment, but no route match was found`)
+      }
+      previousMatch.current = match?.path
+    }
+  }, [location, context, network, metaMetricsId, participateInMetaMetrics])
+
+  /**
+   * track a metametrics event using segment
+   * e.g metricsEvent({ event: 'Unlocked MetaMask', category: 'Navigation' })
+   *
+   * @param {object}  config - configuration object for the event to track
+   * @param {string}  config.event - event name to track
+   * @param {string}  config.category - category to associate event to
+   * @param {boolean} [config.isOptIn] - happened during opt in/out workflow
+   * @param {object}  [config.properties] - object of custom values to track, snake_case
+   * @param {number}  [config.revenue] - amount of currency that event creates in revenue for MetaMask
+   * @param {string}  [config.currency] - ISO 4127 format currency for events with revenue, defaults to US dollars
+   * @param {number}  [config.value] - Abstract "value" that this event has for MetaMask.
+   * @return {undefined}
+   */
+  const trackEvent = useCallback(
+    (config = {}) => {
+      const { event, category, isOptIn = false, properties = {} } = config
+      if (!event) {
+        // Event name is required for tracking an event
+        throw new Error('MetaMetrics trackEvent function must be provided a payload with an "event" key')
+      }
+      if (!category) {
+        // Category must be supplied for every tracking event
+        throw new Error('MetaMetrics events must be provided a category')
+      }
+
+      let excludeMetaMetricsId = config.excludeMetaMetricsId ?? false
+
+      // This is carried over from the old implementation, and will likely need
+      // to be updated to work with the new tracking plan. I think we should use
+      // a config setting for this instead of trying to match the event name
+      const isSendFlow = Boolean(event.match(/^send|^confirm/u))
+      if (isSendFlow && !sendCountIsTrackable(metaMetricsSendCount + 1)) {
+        excludeMetaMetricsId = true
+      }
+      const idTrait = excludeMetaMetricsId ? 'anonymousId' : 'userId'
+      const idValue = excludeMetaMetricsId ? METAMETRICS_ANONYMOUS_ID : metaMetricsId
+
+      if (participateInMetaMetrics || isOptIn) {
+        segment.track({
+          [idTrait]: idValue,
+          event,
+          properties: {
+            ...omit(properties, ['revenue', 'currency', 'value']),
```

Comment:
It's now impossible to send these values even via `config` - nothing seems to be done with them, unless I'm missing something.

We'd want these three values added to `properties` here if defined, right?